tests/integration/metrics/test_integration_metric.py

```diff
@@ -0,0 +1,152 @@
+    lib.entities.google_api._GCP_COMPUTE_ENDPOINT = f"http://localhost:{MOCKED_API_PORT}"
+    lib.entities.extractors.gce_instance._GCP_COMPUTE_ENDPOINT = f"http://localhost:{MOCKED_API_PORT}"
+
+    system_variables.update({
+        'GOOGLE_APPLICATION_CREDENTIALS': f"{resource_path_root}/metrics/token_for_tests.json",
+    })
+
+
+@pytest.fixture(scope="function", autouse=True)
+def setup_config(resource_path_root, mocker: MockerFixture):
+    # change the config dir
+    patched = mocker.patch('os.path.dirname')
+    patched.return_value = os.path.join(resource_path_root, "metrics")
+
+
+@pytest.fixture(scope="function", autouse=True)
+def setup_env(monkeypatch, resource_path_root):
+    for variable_name in system_variables:
+        monkeypatch.setenv(variable_name, system_variables[variable_name])
+
+
+@pytest.fixture(scope="function", autouse=True)
+def clean_journal():
+    Requests.reset_request_journal()
+
+
+@pytest.fixture(scope="session", autouse=True)
+def setup_wiremock():
+    # setup WireMock server
+
+    wiremock = None
+    Config.base_url = 'http://localhost:{}/__admin'.format(MOCKED_API_PORT)
+
+    wiremock = WireMockServer(port=MOCKED_API_PORT)
+    wiremock.start()
+
+    print("wiremock running")
+
+    # run test
+    yield
+
+    # stop WireMock server
+
+    wiremock.stop()
+
+
+@pytest.mark.asyncio
+async def test_metric_authorization_header():
+    await async_dynatrace_gcp_extension()
+
+    request = NearMissMatchPatternRequest(url_path_pattern="/api/v2/metrics/ingest",
+                                          method="POST")
+
+    matched_request: RequestResponseFindResponse = Requests.get_matching_requests(request)
+
+    assert_that(matched_request).is_not_empty()
+
+    result: RequestResponseRequest = matched_request.requests[0]
+    api_key = await get_dynatrace_secret_api_key()
+
+    assert_that(result.headers['Authorization']).is_equal_to(f"Api-Token {api_key}")
+
+
+async def get_dynatrace_secret_api_key():
+    context = LoggingContext(None)
+    async with init_gcp_client_session() as gcp_session:
+        token = await create_token(context, gcp_session)
+        return await fetch_dynatrace_api_key(gcp_session=gcp_session, project_id="dynatrace-gcp-extension",
+                                             token=token)
+
+
+@pytest.mark.asyncio
+async def test_ingest_lines_output(resource_path_root):
+    await async_dynatrace_gcp_extension()
+
+    request = NearMissMatchPatternRequest(url_path_pattern="/api/v2/metrics/ingest",
+                                          method="POST")
+
+    r: RequestResponseFindResponse = Requests.get_matching_requests(request)
+
+    assert_that(r.requests).is_not_empty()
+    result: RequestResponseRequest = r.requests[0]
+
+    body = result.body
+
+    with open(os.path.join(resource_path_root, "metrics/ingest_input.dat")) as ingest:
+        assert_that(ingest.readlines()).is_length(289)
+        ingest.seek(0)
+        recorded_ingest = ingest.read()
```

Comment:
(Minor) you could `readlines` and join them with `\n` instead of reading the file again.

Reply:
Right a good catch.